src/rust/inetstack/protocols/udp/peer.rs

```diff
@@ -241,55 +155,46 @@ impl<const N: usize> SharedUdpPeer<N> {
     pub fn close(&mut self, qd: QDesc) -> Result<(), Fail> {
         #[cfg(feature = "profiler")]
         timer!("udp::close");
-        let queue: SharedUdpQueue<N> = self.runtime.free_queue::<SharedUdpQueue<N>>(&qd)?;
+        trace!("close(): qd={:?}", qd);
+        let mut queue: SharedUdpQueue<N> = self.runtime.free_queue::<SharedUdpQueue<N>>(&qd)?;
         queue.close()?;
         Ok(())
     }
 
     /// Pushes data to a remote UDP peer.
-    pub fn pushto(&mut self, qd: QDesc, data: DemiBuffer, remote: SocketAddrV4) -> Result<(), Fail> {
+    pub fn pushto(&mut self, qd: QDesc, buf: DemiBuffer, remote: SocketAddrV4) -> Result<Pin<Box<Operation>>, Fail> {
         #[cfg(feature = "profiler")]
         timer!("udp::pushto");
-        // Lookup associated endpoint.
+        trace!("pushto(): qd={:?} remote={:?} bytes={:?}", qd, remote, buf.len());
         let mut queue: SharedUdpQueue<N> = self.get_shared_queue(&qd)?;
-        // What happens if not bound?
-        // TODO: Move to a UDP socket state machine.
+        // TODO: Allocate ephemeral port if not bound.
```

Comment:
Could you file an issue for this?

Reply:
Whoops, apparently I already filed one! https://github.com/microsoft/demikernel/issues/973
I will add it to a FIXME